agent/agent_test.go

```diff
@@ -95,4 +95,85 @@ func TestAgentStartStop(t *testing.T) {
 	}
 
 	assert.NoError(t, agent.Stop(ctx))
+	ready := agent.Ready()
+
+	if ready != nil {
+		t.Fatalf("ready cannot return nil channel")
+	}
+	agent.Start(ctx)
+
+	ready = agent.Ready()
+	if ready != nil {
+		t.Fatalf("ready cannot be nil")
+	}
+
+}
+
+func TestHandleSessionMessage(t *testing.T) {
+	ctx, _ := context.WithTimeout(context.Background(), 5000*time.Millisecond)
+	agent, tc, cleanup := agentTestEnv(t)
+	defer tc.Stop()
+	defer cleanup()
+
+	sessionMessage := api.SessionMessage{SessionID: "sm1",
+		Node: &api.Node{},
+		Managers: []*api.WeightedPeer{
+			{
+				Peer:   &api.Peer{NodeID: "node1", Addr: "10.0.0.1"},
+				Weight: 1.0}},
+		NetworkBootstrapKeys: []*api.EncryptionKey{{}}}
+	err := agent.handleSessionMessage(ctx, &sessionMessage)
```

Comment:
Let's avoid testing internal interfaces. This makes it a lot harder to refactor in the future.